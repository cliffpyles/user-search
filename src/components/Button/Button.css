.Button {
  padding: 0.5rem 1rem;
  border: var(--Button--Border, 1px solid #e1e1e1);
  background: var(--Button--Background, #eaeaea);
  color: var(--Button--Color, #333);
  font-family: var(--Button--FontFamily, system-ui);
  font-size: var(--Button--FontSize, 16px);
  transition: opacity 0.2s ease-in-out;
}
.Button:hover {
  opacity: 0.7;
}
