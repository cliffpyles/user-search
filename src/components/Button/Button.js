import React from 'react'
import './Button.css'

export default function Button({ children, onClick, type = 'button' }) {
  return (
    <button className="Button" type={type} onClick={onClick}>
      {children}
    </button>
  )
}
