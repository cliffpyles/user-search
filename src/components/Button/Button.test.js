import React from 'react'
import Renderer from 'react-test-renderer'
import Button from './'

test('can render a Button', () => {
  const component = Renderer.create(
    <Button
      type="submit"
      onClick={() => {
        console.log('clicked')
      }}
    />
  )

  expect(component.toJSON()).toMatchSnapshot()
})
