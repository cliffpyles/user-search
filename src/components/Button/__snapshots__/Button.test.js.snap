// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render a Button 1`] = `
<button
  className="Button"
  onClick={[Function]}
  type="submit"
/>
`;
